### PR TITLE
Add icon/fontawesome support on card component

### PR DESCRIPTION
**Pull request by john-rock**
## What are you changing in this pull request and why?
[Asana Task](https://app.asana.com/0/1200099998847559/1204270295138921/f)

This adds support for icons on Quickstart cards as well as adds support to use both .svg and font awesome icons.

## Preview Links
Quickstart cards - https://deploy-preview-3528--docs-getdbt-com.netlify.app/quickstarts
Docs cards - https://deploy-preview-3528--docs-getdbt-com.netlify.app/

## Additional Info
Loom demo - https://www.loom.com/share/d16839640d1d44b8960052d94cd16655?sid=284882dd-ff61-4bdf-a0bf-4f1b1e94e37a

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Update [Quickstart documentation](https://www.notion.so/dbtlabs/Quickstart-Guides-73c57db4b37547aaab4b7d9437e19eb1) for new icon prop
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

**Comment by john-rock** [comment]
@nghi-ly I added the quickstart icons we had on the [previous version of that page ](https://deploy-preview-3059--docs-getdbt-com.netlify.app/docs/quickstarts/overview). Let me know if you would like to keep them or if we should remove them from this PR.

**Review by JKarlavige — APPROVED** [review]
lgtm! nice quality of life addition here :shipit:

File: website/src/utils/get-icon-type.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Util function to check which icon to render
+import React from "react";
+import { useColorMode } from "@docusaurus/theme-common";
+
+export default function getIconType(icon, ...styles) {
+  const { colorMode } = useColorMode();
+  const combinedStyles = styles.join(" ");
+
+  if (icon.startsWith("fa-")) {
```
Comment (breezyfasano) [inline website/src/utils/get-icon-type.js:9]: this function makes it so easy to use icons, nice strat!

**Review by breezyfasano — APPROVED** [review]
lgtm :shipit:

**Comment by mirnawong1** [comment]
wow, that's soo cool @john-rock ! lgtm as well

Screenshot 2023-06-16 at 15 14 17

**Pull request merged**